ads/google/a4a/utils.js

```diff
@@ -361,6 +354,25 @@ export function additionalDimensions(win, viewportSize) {
 };
 
 /**
+ * Returns a map containing all ad containers that enclose the given ad
+ * element. The params of the map are restricted to the ValidAdContainerTypes,
+ * and the values are all true.
+ * @param {!Element} adElement
+ * @return {!Object<string, boolean>}
+ */
+export function getAllParentAdContainers(adElement) {
```

Comment:
This doesn't necessarily get all parent ad containers, right? Theoretically if there are 20+ passbacks, it wouldn't get them. Is there a better way to break this loop than using 20 as a fixed limit? Maybe instead break when el.parentElement == el.parentElement.parentElement?